### PR TITLE
Fixed resource and shifting tabs

### LINKED ISSUE #8863: Active and Archive Tabs are not properly visible in Shifting Page
**Describe the bug**
When we open the Shifting Page in Desktop, the Tabs: **Active** & **Archive** are not properly visible, making it difficult for the user to switch between them.

**To Reproduce**
1. Go to Facility Page
2. Check Active and Archive Tabs
3. On Desktop view, we can't seamlessly switch between them

**Expected behavior**
The **Active** & **Archive tabs** should appear properly for easy switch between them. 
This can be fixed by adjusting the other elements in the same row

**Screenshots**
This is how it looks on Desktop View ( 100%):

Image

This is how it looks in mobile view:

Image

**Desktop (please complete the following information):**

- OS: Windows
- Browser: Chrome
- Version: Latest

### PR DESCRIPTION
**Pull request by nithish1018**
## Proposed Changes

- Fixes #8863 
- Adjusted Search Input width for Tabs to show up properly
- Added strong types

This is how it looks now:

image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

### CODE REVIEW

File: src/components/Shifting/BoardView.tsx [hunk 1]
```diff
@@ -35,16 +36,16 @@ export default function BoardView() {
   });
 
   const [modalFor, setModalFor] = useState<{
-    externalId?: string;
+    external_id: string | undefined;
```
Comment (Jacobjeevan) [inline src/components/Shifting/BoardView.tsx]: Can keep this as `external_id?: string`.

LGTM.

**Review by Jacobjeevan — APPROVED** [review]

File: src/components/Shifting/BoardView.tsx (hunk 1 above)
Comment (nithishsiliveru) [reply src/components/Shifting/BoardView.tsx]: Added

File: src/components/Shifting/BoardView.tsx [hunk 2]
```diff
@@ -35,16 +36,16 @@ export default function BoardView() {
   });
 
   const [modalFor, setModalFor] = useState<{
-    externalId?: string;
+    external_id?: string | undefined;
```
Comment (shivankacker) [inline src/components/Shifting/BoardView.tsx]:
```suggestion
    external_id?: string;
```
a `?` indirectly means `undefined`

**Review by shivankacker — APPROVED** [review]

File: src/components/Shifting/BoardView.tsx (hunk 2 above)
Comment (nithishsiliveru) [reply src/components/Shifting/BoardView.tsx]: Yes, thanks for the suggestion, I removed it

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

File: src/components/Resource/ResourceBoardView.tsx [hunk 3]
```diff
@@ -67,12 +67,14 @@ export default function BoardView() {
...
-            value={qParams.title}
-            onChange={(e) => updateQuery({ [e.name]: e.value })}
-            placeholder={t("search_resource")}
-          />
+          <div className="w-full md:w-60">
+            <SearchInput
+              name="patient_name"
+              value={qParams.patient_name}
+              onChange={(e) => updateQuery({ [e.name]: e.value })}
+              placeholder={t("search_patient")}
```
Comment (bodhish) [inline src/components/Resource/ResourceBoardView.tsx]: @nithish1018 please self review your code before submitting it for review.

File: src/components/Resource/ResourceBoardView.tsx [hunk 4]
```diff
@@ -67,12 +67,14 @@ export default function BoardView() {
         </div>
 
         <div className="flex w-full flex-col items-center justify-between gap-2 pt-2 xl:flex-row">
-          <SearchInput
-            name="title"
-            value={qParams.title}
-            onChange={(e) => updateQuery({ [e.name]: e.value })}
-            placeholder={t("search_resource")}
-          />
+          <div className="w-full md:w-60">
```
Comment (bodhish) [inline src/components/Resource/ResourceBoardView.tsx]: Can't the search input take these classes as a param? 🤔

File: src/components/Shifting/BoardView.tsx [hunk 5]
```diff
@@ -35,16 +36,16 @@ export default function BoardView() {
   });
 
   const [modalFor, setModalFor] = useState<{
-    externalId?: string;
+    external_id?: string;
```
Comment (bodhish) [inline src/components/Shifting/BoardView.tsx]: Why the casing change? 🤔

**Review by bodhish — CHANGES_REQUESTED** [review]

**Comment by bodhish** [comment]
CC @shivankacker @Jacobjeevan @rithviknishad ☝️

File: src/components/Resource/ResourceBoardView.tsx (hunk 4 above)
Comment (nithishsiliveru) [reply src/components/Resource/ResourceBoardView.tsx]: Thanks for the suggestion, I've added those changes

File: src/components/Shifting/BoardView.tsx (hunk 5 above)
Comment (nithishsiliveru) [reply src/components/Shifting/BoardView.tsx]: The **IShift** interface has that property with lowercase, so I made that change to maintain consistency.
Is it okay or I should revert it back?
image

File: src/components/Resource/ResourceBoard.tsx [hunk 6]
```diff
@@ -71,7 +71,9 @@ export default function BoardView() {
             value={qParams.title}
             onChange={(e) => updateQuery({ [e.name]: e.value })}
             placeholder={t("search_resource")}
+            className="w-full md:w-60"
           />
+
```
Comment (bodhish) [inline src/components/Resource/ResourceBoard.tsx]: Do you need an empty line?

**Review by bodhish — CHANGES_REQUESTED** [review]

File: src/components/Resource/ResourceBoard.tsx (hunk 6 above)
Comment (nithishsiliveru) [reply src/components/Resource/ResourceBoard.tsx]: It has been removed now

**Review by bodhish — APPROVED** [review]

**Comment by bodhish** [comment]
@khavinshankar lgtm

**Pull request merged**